uaclient/tree_ui/_opc_tree_item.py

```diff
@@ -220,10 +220,13 @@ def set_data(
 
         if emit:
             # Emit signal letting subscribers know what data has changed here
-            index = QModelIndex(
-                self.persistent_index(self._ua_column_to_model_column[attribute])
-            )
-            self.data_changed.emit(index, index)
+            try:
+                index = QModelIndex(
+                    self.persistent_index(self._ua_column_to_model_column[attribute])
+                )
+                self.data_changed.emit(index, index)
+            except ValueError:
+                return
```

Comment:
This try/except is wide enough to make me a little uncomfortable: we don't want to mask legit errors here. I would refactor slightly to handle specifically the case we care about, e.g.

```suggestion
            try:
                column = self._ua_column_to_model_column[attribute]
            except ValueError:
                return

            index = QModelIndex(self.persistent_index(column))
            self.data_changed.emit(index, index)
```

Reply:
Ha, see, this is why: this isn't actually the part that is throwing the exception, right? It's the `self.persistent_index`?

Reply:
Ah yes. Good point. Yeah it is the `self.persistent_index` that throws the error. I will make that change.